### PR TITLE
docs(provider/kubernetes): Add a guide for Patch(Manifest) stage.

### PR DESCRIPTION
**Pull request by dibyom**
@lwander @dorbin 

spinnaker/spinnaker#2786

### CODE REVIEW

**Review by lwander — APPROVED** [review]
LGTM -- will wait for a @dorbin review as well

**Comment by dorbin** [comment]
@dibyom 
Now that we have a [pipeline stages reference](https://www.spinnaker.io/reference/pipeline/stages/), you want to create an entry in there for this?

@nhayes FYI

File: guides/user/kubernetes-v2/patch-manifest/index.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  include
+  figure
+  image_path="./in-pipeline.png"
+  caption="Notice that by selecting __Text__ as the __Manifest Source__, we get
+  to enter the manifest YAML by hand."
+%}
+
+### Specify patch content dynamically
+
+Like the [Deploy (Manifest) stage](/guides/user/kubernetes-v2/deploy-manifest/index.md#specify-manifests-dynamically), you can also reference an artifact as the source if you are storing your patch content externally. The artifact must be a text file containing the patch content.
```
Comment (dorbin) [inline guides/user/kubernetes-v2/patch-manifest/index.md:86]: The content of the text file is the same yml format as the statically specified patch above?

File: guides/user/kubernetes-v2/patch-manifest/index.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+containers:
+  - name: my-container
+    image: gcr.io/my-project/my-image:2.0
+```
+For more information on how this works, check out the [binding artifacts docs](/reference/artifacts/in-kubernetes-v2/#binding-artifacts-in-manifests).
+
+
+## Specify Patch Options
+
+You can also optionally specify the following options:
```
Comment (dorbin) [inline guides/user/kubernetes-v2/patch-manifest/index.md:136]: I think you can remove one "option*". I recommend the first one, as "can" implies optional. 
"You can specify the following options:"

File: guides/user/kubernetes-v2/patch-manifest/index.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+* __Record Patch Annotation__
+
+  Defaults to true. When selected, the patch operation including the patch content will be recorded on the patched resource as the `kubernetes.io/change-cause` annotation. If the annotation already exists, the contents are replaced.
+
+
+* __Merge Strategy__
+
+  Three merge strategies are supported for the patch operation:
```
Comment (dorbin) [inline guides/user/kubernetes-v2/patch-manifest/index.md:145]: I think you can delete this sentence, and just list the three.

**Review by dorbin — APPROVED** [review]
Just a few minor things, but lgtm otherwise

File: guides/user/kubernetes-v2/patch-manifest/index.md (hunk 1 above)
Comment (dibyom) [reply guides/user/kubernetes-v2/patch-manifest/index.md:86]: Yes

**Pull request closed without merging**